tiny-firmware/firmware/fsm_bitcoin_impl.c

```diff
@@ -50,26 +51,49 @@
 #include "tiny-firmware/firmware/skyparams.h"
 #include "fsm_bitcoin_impl.h"
 
-ErrCode_t msgBitcoinAddressImpl(BitcoinAddress *msg, ResponseSkycoinAddress *resp) {
-    uint8_t seckey[32] = {0};
-    uint8_t pubkey[33] = {0};
-    uint32_t start_index = !msg->has_start_index ? 0 : msg->start_index;
-    if (!protectPin(true)) {
-        return ErrPinRequired;
-    }
-    if (msg->address_n > 99) {
-        return ErrTooManyAddresses;
-    }
+#define FROMHEX_MAXLEN 512
 
-    if (storage_hasMnemonic() == false) {
-        return ErrMnemonicRequired;
+const uint8_t* fromhex(const char* str)
+{
+    static uint8_t buf[FROMHEX_MAXLEN];
+    size_t len = strlen(str) / 2;
+    if (len > FROMHEX_MAXLEN) len = FROMHEX_MAXLEN;
+    for (size_t i = 0; i < len; i++) {
+        uint8_t c = 0;
+        if (str[i * 2] >= '0' && str[i * 2] <= '9') c += (str[i * 2] - '0') << 4;
+        if ((str[i * 2] & ~0x20) >= 'A' && (str[i * 2] & ~0x20) <= 'F') c += (10 + (str[i * 2] & ~0x20) - 'A') << 4;
+        if (str[i * 2 + 1] >= '0' && str[i * 2 + 1] <= '9') c += (str[i * 2 + 1] - '0');
+        if ((str[i * 2 + 1] & ~0x20) >= 'A' && (str[i * 2 + 1] & ~0x20) <= 'F') c += (10 + (str[i * 2 + 1] & ~0x20) - 'A');
+        buf[i] = c;
     }
+    return buf;
+}
 
-    if (fsm_getKeyPairAtIndex(msg->address_n, pubkey, seckey, resp, start_index, &bitcoin_address_from_pubkey) != ErrOk) {
-        return ErrAddressGeneration;
-    }
-    if (msg->address_n == 1 && msg->has_confirm_address && msg->confirm_address) {
-        return ErrUserConfirmation;
+ErrCode_t msgBitcoinAddressImpl(BitcoinAddress *msg, ResponseSkycoinAddress *resp) {
+
+  if (msg->address_n > 99) {
+      return ErrTooManyAddresses;
+  }
+
+  if (storage_hasMnemonic() == false) {
+      return ErrMnemonicRequired;
+  }
+
+  HDNode* node = fsm_getDerivedNode(SECP256K1_NAME);
+  HDNode addressNode;
+  size_t size_address = 36;
```

Comment:
I think you need to add this constant to bitcoin_constants.h

Reply:
Ok, have fixed it